esmvalcore/preprocessor/_volume.py

```diff
@@ -158,7 +158,9 @@ def calculate_volume(cube: Cube) -> da.core.Array:
     # Calculate Z-direction thickness
     thickness = depth.core_bounds()[..., 1] - depth.core_bounds()[..., 0]
     if cube.has_lazy_data():
-        thickness = da.array(thickness)
+        thickness = da.array(thickness).rechunk(
+            tuple(cube.lazy_data().chunks[d] for d in z_dim)
+        )
```

Comment:
It is better to avoid rechunking if not necessary. Therefore I would recommend doing something like:
```suggestion
        chunks = tuple(cube.lazy_data().chunks[d] for d in z_dim)
        if isinstance(thickness, da.Array):
            thickness = thickness.rechunk(chunks)
        else:
            thickness = da.asarray(thickness, chunks=chunks)
        )
```